src/pages/instances/actions/PublishInstanceBtn.tsx

```diff
@@ -0,0 +1,47 @@
+import { FC } from "react";
+import { LxdInstance } from "types/instance";
+import { ActionButton } from "@canonical/react-components";
+import usePortal from "react-useportal";
+import CreateImageFromInstanceForm from "../forms/CreateImageFromInstanceForm";
+
+interface Props {
+  instance: LxdInstance;
+  isDeleting: boolean;
+  isStopping: boolean;
+}
+
+const PublishInstanceBtn: FC<Props> = ({
+  instance,
+  isDeleting,
+  isStopping,
+}) => {
+  const { openPortal, closePortal, isOpen, Portal } = usePortal();
+
+  return (
+    <>
+      {isOpen && (
+        <Portal>
+          <CreateImageFromInstanceForm
+            close={closePortal}
+            instance={instance}
+          />
+        </Portal>
+      )}
+      <ActionButton
+        appearance="default"
+        onClick={openPortal}
+        aria-label="Publish instance"
+        title="Publish instance"
+        disabled={
+          instance?.status == ("Running" || "Restarting") ||
+          !isStopping ||
```

Comment:
instead doing these `| |` conditions, rather define a list with applicable instance statuses and do something like `statuses.includes(instance.status)`, you should come up with a better variable naming for the list though. The logic `!isStopping` here makes `instance?.status == ("Running" || "Restarting")` redundant btw.

Reply:
If isStopping covers Running, Restarting and Frozen (as it seems to be in my case), then surely the only cases that need to be addressed are isStopping and isDeleting.

Reply:
given and instance may be in the below states:
```
export type LxdInstanceStatus =
  | "Error"
  | "Freezing"
  | "Frozen"
  | "Restarting"
  | "Running"
  | "Starting"
  | "Stopped"
  | "Stopping";
  ```
  Your `!isStopping` condition would disable the button for every other status excepted for `Stopped`. This could be correct but we don't know if `Error`, `Frozen`, `Running` allows for image publish. I tried publishing an image while the instance is running and it worked, not sure about `Error` and `Frozen`. For `Freezing`, `Restarting`, `Starting` and `Stopping` we should disable the button since those are transient states for the instance